lib/public/AppFramework/Http/Response.php

```diff
@@ -231,7 +236,9 @@ public function addHeader($name, $value) {
 
 	/**
 	 * Set the headers
-	 * @param array $headers value header pairs
+	 * @template NewH as array<string, mixed>
+	 * @param NewH $headers value header pairs
+	 * @psalm-this-out self<S, NewH>
```

Comment:
```suggestion
	 * @psalm-this-out static<S, NewH>
```
Using that does not avoid the need for the duplication in subclasses?

Reply:
That's actually something I oversaw, maybe this can make it work.

Reply:
Yep this works! No need to override the methods (except for deprecating setStatus in some places).